Automattic/woocommerce-payments
# Replace E2E Server docker instance with live

Fixes #3293 

### Changes proposed in this Pull Request

Based on the discussion on this thread paJDYF-2PM-p2#comment-8617, running WCPay Server instance in GH Actions has security & latency issues. 

Changes:
* Skips creation of WCPay server docker instance for running E2E Tests and uses Live server on GH Actions. 
* Allows running WCPay server docker instance locally along with the option to use Live server via declaring local env variables.
* E2E Tests Documentation update

### Testing instructions

**Testing GH Actions**

* Go to https://github.com/Automattic/woocommerce-payments/actions/workflows/e2e-test.yml
* Click `Run workflow` and choose the branch `e2e-optim/use-live-server` to trigger the E2E tests manually
* Choose the created test suite -> Choose test -> Look for the step `Run npm run test:e2e`
* Confirm that the server instance is not created. [Old log](https://user-images.githubusercontent.com/1161872/141801626-8481a1c9-dd28-4959-98b4-0024a2790670.png) | [New log](https://user-images.githubusercontent.com/1161872/141801918-6258b342-bb7a-4427-910a-1b38258079ae.png)

**Testing Local**

* Following the instructions in this [README.pdf](https://github.com/Automattic/woocommerce-payments/files/7603465/README.pdf) (Copy of the documentation) and try running local server & live server. 
* Confirm that the changes works as expected.

-------------------


**Post merge**

* Remove repo secrets related to WCPay Server setup (@achyuthajoy)
